packages/core/src/components/SearchBox/SearchBox.tsx

```diff
@@ -0,0 +1,29 @@
+import { CloseIcon, SearchIcon } from '@medly-components/icons';
+import { useCombinedRefs, WithStyle } from '@medly-components/utils';
+import React, { SFC, useState } from 'react';
+import * as Styled from './SearchBox.styled';
+import { Props } from './types';
+
+export const SearchBox: SFC<Props> & WithStyle = React.memo(
+    React.forwardRef((props, ref) => {
+        const inputRef = useCombinedRefs<HTMLInputElement>(ref, React.useRef(null)),
+            [showCloseIcon, setCloseIconState] = useState(false);
+
+        const handleFocus = () => setCloseIconState(true);
```

Comment:
Wrap both if these handlers in useCallback. So that the consumer of this wrapper will not render again n again.